admin/worker/run_autoscaler.go

```diff
@@ -88,7 +99,11 @@ func (w *Worker) allRecommendations(ctx context.Context) ([]metrics.AutoscalerSl
 			break
 		}
 		recs = append(recs, batch...)
-		offset += limit
+		if len(batch) >= limit {
+			offset += limit
+		} else {
+			break
+		}
```

Comment:
Here you can also reduce nesting by changing to something like:
```suggestion
		if len(batch) < limit {
			break
		}
		offset += limit
```

Reply:
👍🏼